pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,41 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, certificate *corev1.LocalObjectReference, clusterDomain string) error {
+	clusterHasManagedDomain := clusterHasManagedDomain(clusterDomain)
+
+	if certificate == nil {
+		if clusterHasManagedDomain {
+			return fmt.Errorf("ARO has not provisioned the default ingress certificate because it is empty and cluster has a managed domain")
+		}
 		return errNoDefaultCertificate
 	}
 
-	if ingress.Spec.DefaultCertificate.Name != string(cv.Spec.ClusterID)+"-ingress" {
-		return fmt.Errorf("custom ingress certificate in use: %q", ingress.Spec.DefaultCertificate.Name)
+	certificateName := certificate.Name
+	certificateNameIsValid := certificateNameIsValid(certificateName, clusterId)
+
+	if clusterHasManagedDomain && !certificateNameIsValid {
+		return fmt.Errorf("custom ingress certificate in use: %q", certificateName)
 	}
 
 	return nil
 }
+
+func certificateNameIsValid(certificateName string, clusterID configv1.ClusterID) bool {
+	ingressNameSuffix := "-ingress"
+	expectedCertName := string(clusterID) + ingressNameSuffix
+	return certificateName == expectedCertName
+}
+
+func clusterHasManagedDomain(clusterDomain string) bool {
+	managedDomainSuffix := "aroapp.io"
+	return strings.HasSuffix(clusterDomain, managedDomainSuffix)
```

Comment:
the variable isn't necessary here, at least not at this scope. maybe we should have a const or just `return strings.HasSuffix(clusterDomain, "aroapp.io")`

Reply:
I don't like magic hardcoded strings and as the string is just used once, declaring the variable inside the short function was enough for me.
Going for the const option now as you suggest  😄

Note that I have also moved it to the dns package and tested the public function so it is reusable.
Tks!